File: src/stack/lc_232_implement_queue_using_stacks/implement_queue_using_stacks.py
class MyQueue:

    def __init__(self):
        self.s1 = []
        self.s2 = []

    def push(self, x: int) -> None:
        self.s1.append(x)

    def pop(self) -> int:
        while len(self.s1) > 1:
            self.s2.append(self.s1.pop())
        
        t = self.s1.pop()
        while self.s2:
            self.s1.append(self.s2.pop())
        return t

    def peek(self) -> int:
        while len(self.s1) > 1:
            self.s2.append(self.s1.pop())
        
        t = self.s1[0]
        while self.s2:
            self.s1.append(self.s2.pop())
        return t

    def empty(self) -> bool:
        return len(self.s1) == 0

File: src/stack/lc_232_implement_queue_using_stacks/implement_queue_using_stacks.java
import java.util.*;
class MyQueue {

    Deque<Integer> front;

    public MyQueue() {
        front = new LinkedList<>();
    }
    
    public void push(int x) {
        front.addLast(x);
    }
    
    public int pop() {
        return front.removeFirst();
    }
    
    public int peek() {
        return front.peek();
    }
    
    public boolean empty() {
        return front.isEmpty();
        
    }
}
